functions/unittest/parseAirtable.test.js

```diff
@@ -0,0 +1,70 @@
+const mockAirtable = require ('../airtable-shared-view.js');
+const { parseAirtableData } = require('../airtable-shared-view.js');
+
+describe('mapColumn() should detect any ftm-name in [] and set up the appropriate translator based on column type', () => {
+  const testColumn = [
+    {
+      name: 'This string should not be matched [matched-string]',
+      type: 'text',
+      id: 'first',
+    },
+    {
+      name: 'Nothing should match here',
+      type: 'select',
+      id: 'second',
+    },
+  ];  
+  test('If the column name has a string encapsulated in [], regex should match and a ftm-name property should be made', () => {
+    const copyTestColumn = JSON.parse(JSON.stringify(testColumn));
+    const returnColumns = mockAirtable.unitTestFunctions.mapColumns(copyTestColumn);
+    expect(returnColumns.first).toBeDefined();
+    expect(returnColumns.first.ftm_name).toEqual('matched-string');
+    expect(returnColumns.second.ftm_name).toBeUndefined();
+  });
+  test('There should be a default function for valueTranslator, specific functions should be used to match the column type', () => {
+    const copyTestColumn = JSON.parse(JSON.stringify(testColumn));
+    const returnColumns = mockAirtable.unitTestFunctions.mapColumns(copyTestColumn);
+    expect(returnColumns.first.valueTranslator).toEqual(mockAirtable.unitTestFunctions.identityTranslator);
+    expect(returnColumns.second.valueTranslator).not.toEqual(mockAirtable.unitTestFunctions.identityTranslator);
+    expect(returnColumns.second.valueTranslator).toEqual(expect.any(Function));
+  });
+});
+
+describe('parseAirtableData() should be able to go through each row and generate usable data structure for us', () => {
+  const testData = {
+    data: {
+      table: {
+        columns: [
+          {
+            name: 'This string should not be matched [matched-string]',
+            type: 'text',
+            id: 'first',
+          },
+          {
+            name: 'Nothing should match here',
+            type: 'text',
+            id: 'second',
+          },
+        ],
+        rows: [
+          {
+            id: 1,
+            cellValuesByColumnId: {
+              first: ' this string has an intial space to be trimmed by default valueTranslator',
```

Comment:
this test seems to be testing a few different things
- that the data structure is created correctly
- that the valueTranslator trims whitespace

I think that you could split this up to test the parseAirtableData function is returning the correct structure and then have different tests for the logic within each translator

Reply:
Ok, I did not actually go ahead and write tests about the functionality for the different valueTranslators() because I wasn't sure if they are up for change or not. I'll separate the test for the correct data structure and create some new tests for the functionalities of the valueTranslator().